### PR TITLE
Fix user permissions for custom page controllers

### LINKED ISSUE #3431: Page controllers do not show up in 'allowed page types'
**Affected version(s)**

Contao Contao 4.12

**Description**

Page types that were created via controllers + the `@Page` annotation don't show up in the privilege settings for page types:

Screenshot_20210907_123619

### PR DESCRIPTION
**Pull request by aschempp**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #3431

### CODE REVIEW

**Review by Toflar — APPROVED** [review]

**Review by fritzmg — APPROVED** [review]

File: core-bundle/src/EventListener/DataContainer/PageTypeOptionsListener.php [hunk 1]
```diff
@@ -51,6 +53,10 @@ public function __invoke(DataContainer $dc): array
     {
         $options = array_unique(array_merge(array_keys($GLOBALS['TL_PTY']), $this->pageRegistry->keys()));
 
+        if ('tl_user' === $dc->table || 'tl_user_group' === $dc->table) {
```
Comment (leofeyer) [inline core-bundle/src/EventListener/DataContainer/PageTypeOptionsListener.php:56]: Should this not be added after the event has been dispatched?

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: core-bundle/src/EventListener/DataContainer/PageTypeOptionsListener.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/EventListener/DataContainer/PageTypeOptionsListener.php:56]: No. The event is meant to limit the page options (e.g. Only show the root option on pid=0). That does not apply here.

**Comment by m-vo** [comment]
The changes from this PR are working. :tada:  

However, the translations are missing in the checkbox widget (only shows the key).

**Comment by aschempp** [comment]
> However, the translations are missing in the checkbox widget (only shows the key).

For all options or only for the custom pages controllers? Did you add labels to `$GLOBALS['TL_LANG']['PTY']`?

**Comment by m-vo** [comment]
> For all options or only for the custom pages controllers? 

Only for the custom page controllers.

> Did you add labels to $GLOBALS['TL_LANG']['PTY']?

Yes.

**Review by leofeyer — APPROVED** [review]

**Pull request merged**